x/tokenomics/keeper/token_logic_modules.go

```diff
@@ -0,0 +1,443 @@
+	// on behalf of the application account.
+	if err = k.bankKeeper.BurnCoins(
+		ctx, apptypes.ModuleName, sdk.NewCoins(settlementCoins),
+	); err != nil {
+		return tokenomicstypes.ErrTokenomicsApplicationModuleBurn.Wrapf("burning %s from the application module account: %v", settlementCoins, err)
+	}
+	logger.Info(fmt.Sprintf("burned (%v) from the application module account", settlementCoins))
+
+	// Update the application's on-chain stake
+	newAppStake, err := application.Stake.SafeSub(settlementCoins)
+	if err != nil {
+		return tokenomicstypes.ErrTokenomicsApplicationNewStakeInvalid.Wrapf("application %q stake cannot be reduced to a negative amount %v", application.Address, newAppStake)
+	}
+	application.Stake = &newAppStake
+	logger.Info(fmt.Sprintf("updated application %q stake to %v", application.Address, newAppStake))
+
+	return nil
+}
+
+// TokenLogicModuleGlobalMint processes the business logic for the GlobalMint TLM.
+// TODO_UPNEXT(@olshansk): Delete this in favor of a real TLM that mints tokens
+// and distributes them to the appropriate accounts via boosts.
+func (k Keeper) TokenLogicModuleGlobalMint(
+	ctx context.Context,
+	service *sharedtypes.Service,
+	application *apptypes.Application,
+	supplier *sharedtypes.Supplier,
+	settlementCoins cosmostypes.Coin,
+	relayMiningDifficulty *tokenomictypes.RelayMiningDifficulty,
+) error {
+	logger := k.Logger().With("method", "TokenLogicModuleGlobalMint")
+
+	// Determine how much new uPOKT to mint based on global inflation
+	settlementAmtFloat := new(big.Float).SetUint64(settlementCoins.Amount.Uint64())
+	newMintAmtFloat := settlementAmtFloat.Mul(settlementAmtFloat, big.NewFloat(MintGlobalAllocation))
+	newMintAmtInt, _ := newMintAmtFloat.Int64()
+	newMintCoins := sdk.NewCoins(cosmostypes.NewCoin(volatile.DenomuPOKT, math.NewInt(newMintAmtInt)))
+
+	// Mint new uPOKT to the tokenomics module account
+	if err := k.bankKeeper.MintCoins(ctx, tokenomictypes.ModuleName, newMintCoins); err != nil {
+		return tokenomicstypes.ErrTokenomicsModuleMintFailed.Wrapf(
+			"minting %s to the tokenomics module account: %v", newMintCoins, err)
+	}
+	logger.Info(fmt.Sprintf("minted (%v) coins in the tokenomics module", newMintCoins))
+
+	// Send a portion of the rewards to the application
+	if err := k.sendRewardsToAccount(ctx, application.Address, settlementAmtFloat, MintAllocationApplication); err != nil {
+		return err
+	}
+
+	// Send a portion of the rewards to the supplier
+	if err := k.sendRewardsToAccount(ctx, supplier.Address, settlementAmtFloat, MintAllocationSupplier); err != nil {
+		return err
+	}
+
+	// Send a portion of the rewards to the DAO
+	if err := k.sendRewardsToAccount(ctx, k.GetAuthority(), settlementAmtFloat, MintAllocationDAO); err != nil {
+		return err
+	}
+
+	// Send a portion of the rewards to the source owner
+	if err := k.sendRewardsToAccount(ctx, service.OwnerAddress, settlementAmtFloat, MintAllocationSourceOwner); err != nil {
+		return err
+	}
+
```

Comment:
Can we log any remaining minted but not distributed rewards here? As it may happen due to float rounding.

Reply:
Done!